lib/matplotlib/markers.py

```diff
@@ -385,11 +418,83 @@ def get_alt_transform(self):
         Return the transform to be applied to the `.Path` from
         `MarkerStyle.get_alt_path()`.
         """
-        return self._alt_transform.frozen()
+        return _fast_transform_combine(self._alt_transform,
+                                       self._user_transform)
 
     def get_snap_threshold(self):
         return self._snap_threshold
 
+    def get_user_transform(self):
+        """Return user supplied part of marker transform."""
+        if self._user_transform is not None:
+            return self._user_transform.frozen()
+
+    def transformed(self, transform: Affine2D):
+        """
+        Return new marker with combined transformation.
+
+        Parameters
+        ----------
+        transform : Affine2D, default: None
+            Transform will be combined with current user supplied transform.
+        """
+        new_marker = MarkerStyle(self)
+        if new_marker._user_transform is not None:
+            new_marker._user_transform += transform
+        else:
+            new_marker._user_transform = transform
+        return new_marker
+
+    def rotated(self, deg=None, rad=None):
+        """
+        Return new marker rotated by specified angle.
+
+        Parameters
+        ----------
+        deg : float, default: None
+            Use this parameter to specify rotation angle in degrees.
+
+        rad : float, default: None
+            Use this parameter to specify rotation angle in radians.
+
+        .. note:: You must specify exactly one of deg or rad.
+        """
+        if not (deg is None) ^ (rad is None):
+            raise ValueError("Exactly one of deg or rad shall be used.")
+
```

Comment:
```suggestion
        if deg is None and rad is None:
            raise ValueError('One of deg or rad is required')
        if deg is not None and rad is not None:
            raise ValueError('Only one of deg and rad can be supplied')
```

I had to look up xor, and in my opinion the error is more clear if you differentiate the two cases.